### PR TITLE
Check bonds for 3D inputs and auto generate

### PR DESCRIPTION
**Pull request by psavery**
This checks the bonding for 3D inputs to see if it either doesn't
have bonds or only has single bonds. If either of these are true,
it will automatically attempt to detect the bonds. This fixes cases
where molecules like caffeine, for instance, are reported as having
all single bonds. It is very important for us to get the bonding
correct, or else the InChI and InChI key will not be correct.

Fixes: #147

### CODE REVIEW

**Comment by psavery** [comment]
@cryos This currently autodetects bonds if either no bond orders are provided, or all bond orders are 1.

I can add in some other checks if you think that would be good. For instance:

1. Check to see if more than one atom is at (0, 0, 0). Don't autodetect bonds if that's the case.
2. Check all atom-atom distances to see if any are less than a tolerance. Don't autodetect bonds if that is the case.

File: girder/molecules/molecules/utilities/molecules.py [hunk 1]
```diff
@@ -32,10 +34,17 @@ def create_molecule(data_str, input_format, user, public, gen3d=True,
...
 
-    if using_2d_format or input_format in openbabel_3d_formats:
+    if using_2d_format:
         inchi, inchikey = openbabel.to_inchi(data_str, input_format)
     else:
-        sdf_data = avogadro.convert_str(data_str, input_format, 'sdf')
+        # Let's make sure the bonds look reasonable
+        cjson = convert_3d_format_to_cjson(data_str, input_format)
+        if bonding_looks_suspicious(cjson):
+            tmp = openbabel.autodetect_bonds(cjson)
```
Comment (cjh1) [inline girder/molecules/molecules/utilities/molecules.py]: Is `autodetect_bonds` computationally intensive? Should it be done asynchronously?

Comment (psavery) [reply girder/molecules/molecules/utilities/molecules.py]: I don't think it is, but I will give it a try on a big molecule to see how long it takes. I think it essentially loops through all atom-atom pairs and calculates the distance between them, and determines if the their distance is less than the sum of their covalent radii, or something like that.

Comment (psavery) [reply girder/molecules/molecules/utilities/molecules.py]: I gave it a try on a real molecule with 457 atoms (207 heavy atoms), and the `autodetect_bonds()` function there took about 0.04 seconds on my computer.

Comment (cjh1) [reply girder/molecules/molecules/utilities/molecules.py]: Not an issue then :-)

**Comment by cjh1** [comment]
@psavery Can you resolve this conflict?

**Pull request merged**